django_mongodb/compiler.py

```diff
@@ -23,14 +23,18 @@ def execute_sql(
         # QuerySet.count()
         if self.query.annotations == {"__count": Count("*")}:
             return [self.get_count()]
-        # Specify columns if there are any annotations so that annotations are
-        # computed via $project.
-        columns = self.get_columns() if self.query.annotations else None
```

Comment:
I think we lost an optimization here where `$project` is omitted unless needed (i.e. there are annotations).

For example, the first query in `basic.tests.ModelLookupTest.test_equal_lookup` went from:
```
basic_article.aggregate([{'$match': {'$expr': {'$eq': ['$_id', ObjectId('668da533d38a1c8d3ee7708b')]}}}, {'$limit': 21}]);`
```
to
```
basic_article.aggregate([{'$match': {'$expr': {'$eq': ['$_id', ObjectId('668da57447faa4a8f994ebd1')]}}}, {'$project': {'_id': 1, 'headline': 1, 'pub_date': 1}}, {'$limit': 21}]);
```
We could fix it by making `self.get_columns() if self.query.annotations else None` to `build_query()` but there might be a better place to put it.

Adding a test to the Django fork for this at some point would probably be a good idea.

Reply:
Yes, I added this new layer thinking that is not a big deal. If I change the code a bit, the old behavior is preserved. So, I will change this code in order to keep the previous behavior.

Reply:
I'm not sure how much it matters performance-wise, but particularly when debugging queries, I think it's cleaner not to include $project if it's not needed, as long as the solution for doing so doesn't get super complicated.

Reply:
it's not. As you mention above, we need to pass a None to build query if not self.query.annotation. Running the test, will update soon.